nuitka/nodes/IterationHandles.py

```diff
@@ -0,0 +1,310 @@
+
+    Methods
+    -------
+    getIterationValueWithIndex(value_index)
+        Sequential access of the constants
+    """
+
+    def __init__(self, constant_node):
+        ConstantIterationHandleBase.__init__(self, constant_node)
+        assert type(self.constant) not in (set, dict)
+
+    def getIterationValueWithIndex(self, value_index):
+        """Tries to return constant value at the given index.
+
+        Parameters
+        ----------
+        value_index : int
+            Index value of the element to be returned
+        """
+        try:
+            from .ConstantRefNodes import makeConstantRefNode
+
+            return makeConstantRefNode(
+                constant=self.constant[value_index],
+                source_ref=self.constant_node.source_ref,
+            )
+        except IndexError:
+            return None
+
+
+class ConstantSetAndDictIterationHandle(ConstantIterationHandleBase):
+    """Class for the set and dictionary constants.
+    """
+
+    def __init__(self, constant_node):
+        ConstantIterationHandleBase.__init__(self, constant_node)
+        assert type(self.constant) in (set, dict)
+
+
+class ListAndTupleContainerMakingIterationHandle(IterationHandleBase):
+    """Class for list and tuple container making expression
+
+    Attributes
+    ----------
+    constant_node : node_object
+        Instance of the calling node.
+
+    Methods
+    -------
+    __repr__()
+        Prints representation of the ListAndTupleContainerMakingIterationHandle
+        object
+    getNextValueExpression()
+        Returns the next iteration value
+    getNextValueTruth()
+        Returns the boolean value of the next handle
+    getIterationValueWithIndex(value_index)
+        Sequential access of the expression
+    """
+
+    def __init__(self, constant_node):
+        assert type(constant_node) in (list, tuple)
+        self.constant_node = constant_node
+        self.iter = iter(constant_node)
+
+    def __repr__(self):
+        return "<%s of %r>" % (self.__class__.__name__, self.constant_node)
+
+    def getNextValueExpression(self):
+        """Return the next iteration value or StopIteration exception
+        if the iteration has reached the end
+        """
+        try:
+            return next(self.iter)
+        except StopIteration:
+            return None
+
+    def getIterationValueWithIndex(self, value_index):
+        """Tries to return constant value at the given index.
+
+        Parameters
+        ----------
+        value_index : int
+            Index value of the element to be returned
+        """
+        try:
+            return self.constant_node[value_index]
+        except IndexError:
+            return None
+
+    def getNextValueTruth(self):
+        """Return the boolean value of the next iteration handle."""
+        try:
+            iteration_value = next(self.iter)
+        except StopIteration:
+            return StopIteration
+        return bool(iteration_value)
```

Comment:
The type of `iteration_value` is not what you treat it as here. It is a node, and should be asked its truth value. This is going to be `true` in this form always.